### DISCUSSION TITLE
Approach to reviewing IATI Publisher's extra validation rules

### DISCUSSION THREAD

**Discussion by emmajclegg** [younginnovations/iatipublisher#1451] [General]
We understand that IATI Publisher applies its own IATI data checks, additional to those from the IATI Validator. It would be good to discuss how easy it is for these to be removed or updated in IATI Publisher's code.

This was first highlighted in this support ticket (#[43402](https://iati.zendesk.com/agent/tickets/43402)) where YI explained that:
_"The IATI Validator api check is performed only when the activity is being published to the IATI Registry, in other cases, the validation present is not directly linked to IATI Validator but is dependent on the IATI Publisher system itself."_

Example additional checks:
- A "revised" activity budget should have an equivalent "original" budget for the same period
`Budget with type revised must have period start and end same to that of one of the budgets having same type original for budgets elements at position`
- The narrative field of recipient-country should have "language" populated (even if the narrative sub-element is unpopulated)
`Recipient country: The narrative field is required with @xml:lang field.`
- Actual results values are expected if a target is present (even if the result's end date is in the future)
`Period: x-y (Target value is present here, actual value is missing)`

Open questions:
- Is there a standalone list of all of these additional validation rules, or are they hard-coded across different IATI Publisher code scripts?
- When are these checks applied - at the point users are trying to enter and save data in IATI Publisher?
- Is there background to how these rules were chosen?

**Comment by emmajclegg** [comment]
@praweshsth @Sanilblank - I've started this as a discussion. It's not pressing but would be useful to understand better (see the questions above)

**Comment by praweshsth** [comment]
@emmajclegg 

Is there a standalone list of all of these additional validation rules, or are they hard-coded across different IATI Publisher code scripts?
**- Hard coded across different IATI code** 

When are these checks applied - at the point users are trying to enter and save data in IATI Publisher?
**- When saving the data**

Is there background to how these rules were chosen?
**-We heavily depended on the IATI Standard that is present in https://iatistandard.org/en/iati-standard/203/activity-standard/**
